yarn-project/sequencer-client/src/publisher/config.ts

```diff
@@ -1,41 +1,53 @@
-import { L1ContractAddresses } from '@aztec/ethereum';
+import { NULL_KEY, l1ContractAddresses } from '@aztec/ethereum';
+import { stringlyNumber, z } from '@aztec/foundation/zod';
 
-/**
- * The configuration of the rollup transaction publisher.
- */
-export interface TxSenderConfig {
+export const txSenderConfig = z.object({
   /**
    * The private key to be used by the publisher.
    */
-  publisherPrivateKey: `0x${string}`;
+  publisherPrivateKey: z
+    .string()
+    .regex(/^0x[0-9a-f]+$/i)
+    .optional()
+    .default(NULL_KEY)
+    .transform((val): `0x${string}` => {
+      return val.startsWith('0x') ? (val as `0x${string}`) : `0x${val}`;
+    }),
 
   /**
    * The RPC Url of the ethereum host.
    */
-  rpcUrl: string;
+  rpcUrl: z.string().optional().default(''),
```

Comment:
Not sure if you were aware, zod does have a url validator. Saw this comment on github
```
z.string().url().optional().or(z.literal(''))
```

Reply:
Oh that's great. I hadn't considered enforcing it be an url